linkerd/tls/route/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+//! An TLS route matching library for Linkerd to support the TLSRoute
+//! Kubernetes Gateway API types.
+
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+use linkerd_tls::ServerName;
+use r#match::SessionMatch;
+use tracing::trace;
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct SessionInfo {
+    pub sni: ServerName,
+}
```

Comment:
This should be moved down below the mods with the other structs. I would probably move it to directly about the `find` impl, so that it's defined nearest where it is used.

It would also be good to include a comment a brief comment describing that this is intended to capture the routing metadata of a proxied/unterminated server-side TLS connection.